kernel/src/net/socket/util/options.rs

```diff
@@ -26,11 +32,102 @@ impl SocketOptionSet {
             sock_errors: None,
             reuse_addr: false,
             reuse_port: false,
-            send_buf: SEND_BUF_LEN as u32,
-            recv_buf: RECV_BUF_LEN as u32,
+            send_buf: TCP_SEND_BUF_LEN as u32,
+            recv_buf: TCP_RECV_BUF_LEN as u32,
             linger: LingerOption::default(),
         }
     }
+
+    /// Return the default socket level options for udp socket.
+    pub fn new_udp() -> Self {
+        Self {
+            sock_errors: None,
+            reuse_addr: false,
+            reuse_port: false,
+            send_buf: UDP_SEND_PAYLOAD_LEN as u32,
+            recv_buf: UDP_RECV_PAYLOAD_LEN as u32,
+            linger: LingerOption::default(),
+        }
+    }
+
+    /// Get option. Return Ok(true) if it need to clear the socket errors
+    pub fn get_option(&self, option: &mut dyn SocketOption) -> Result<bool> {
+        // Note that the socket error has to be handled separately, because it is automatically
+        // cleared after reading.
+        match_sock_option_mut!(option, {
+            socket_errors: SocketError => {
+                let sock_errors = self.sock_errors();
+                socket_errors.set(sock_errors);
+
+                return Ok(true);
+            },
+            _ => ()
+        });
+
+        match_sock_option_mut!(option, {
+            socket_reuse_addr: ReuseAddr => {
+                let reuse_addr = self.reuse_addr();
+                socket_reuse_addr.set(reuse_addr);
+            },
+            socket_send_buf: SendBuf => {
+                let send_buf = self.send_buf();
+                socket_send_buf.set(send_buf);
+            },
+            socket_recv_buf: RecvBuf => {
+                let recv_buf = self.recv_buf();
+                socket_recv_buf.set(recv_buf);
+            },
+            socket_reuse_port: ReusePort => {
+                let reuse_port = self.reuse_port();
+                socket_reuse_port.set(reuse_port);
+            },
+            socket_linger: Linger => {
+                let linger = self.linger();
+                socket_linger.set(linger);
+            },
+            _ => return_errno_with_message!(Errno::ENOPROTOOPT, "the socket option to get is unknown")
+        });
+        Ok(false)
+    }
+
+    pub fn set_option(&mut self, option: &dyn SocketOption) -> Result<()> {
+        // FIXME: here we have only set the value of the option, without actually
+        // making any real modifications.
+        match_sock_option_ref!(option, {
+            // Socket options:
```

Comment:
This comment is not needed anymore. After all, all options in this file are socket level options.